### PR TITLE
Replace os-3.11 with okd-4.1 provider at CI

### PR DESCRIPTION
**Pull request by qinqon**
It has a working version of secondary nics on multinode

### CODE REVIEW

**Comment by qinqon** [comment]
ci test please

**Comment by phoracek** [comment]
OKD fails with:

```
IMAGE_REGISTRY=localhost:failed to found the container with name nmstate_kubernetes-nmstate_standard-check-pr-dnsmasq \
			   make push-handler
/bin/sh: to: command not found
```

File: hack/install-kubevirtci.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash -e
 
 organization=kubevirt
-commit="1c2922cd9ebaabf7006dc68aa8df70ad84d96d4b"
+commit="22a492a3de940bc1972bf66be13e836accc4533c"
```
Comment (oshoval) [inline hack/install-kubevirtci.sh]: i think kubevirtci has new image for okd 4.1

db8c24bf830bb927f01829e6c9f083627fe6b832

Comment (phoracek) [reply hack/install-kubevirtci.sh]: Thanks Or. I think this is on hold anyways, until we get multi-nic support in OKD.

Comment (qinqon) [reply hack/install-kubevirtci.sh]: Yep, without multinic at okd4 we cannot test this.

**Comment by phoracek** [comment]
@qinqon I'm working on a PR bumping version of Kubernetes and get the same problem:

```
IMAGE_REGISTRY=localhost:port is not exposed \
                           make push-handler
/bin/sh: is: command not found
```

File: Makefile [hunk 2]
```diff
@@ -124,24 +123,27 @@ cluster-clean: $(KUBECTL)
 	$(KUBECTL) delete --ignore-not-found -f deploy/
 	$(KUBECTL) delete --ignore-not-found -f deploy/crds/nmstate_v1alpha1_nodenetworkstate_crd.yaml
 	$(KUBECTL) delete --ignore-not-found -f deploy/crds/nmstate_v1alpha1_nodenetworkconfigurationpolicy_crd.yaml
-	if [[ "$$KUBEVIRT_PROVIDER" =~ ^os-.*$$ ]]; then \
+	if [[ ! "$$KUBEVIRT_PROVIDER" =~ ^k8s-.*$$ ]]; then \
```
Comment (phoracek) [inline Makefile]: Negative checks are evil, please stay checking for okd-. The fact that something is not kubernetes does not mean it is openshift ;)

File: hack/install-nm.sh [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-docker exec $node01_id ssh.sh sudo yum install -y NetworkManager
-docker exec $node01_id ssh.sh sudo systemctl daemon-reload
-docker exec $node01_id ssh.sh sudo systemctl restart NetworkManager
-
-echo 'Check NetworkManager is working fine'
-docker exec $node01_id ssh.sh nmcli device show > /dev/null
+script_dir=$(dirname "$(readlink -f "$0")")
+ssh=$script_dir/../kubevirtci/cluster-up/ssh.sh
+
+if [[ ! "$KUBEVIRT_PROVIDER" =~ okd ]]; then
```
Comment (phoracek) [inline hack/install-nm.sh]: Again, please do positive check please.

File: hack/install-nm.sh [hunk 4]
```diff
@@ -1,13 +1,15 @@
 #!/bin/bash -e
 
-# TODO: Iterate all the nodes
```
Comment (phoracek) [inline hack/install-nm.sh]: This comment is still relevant

File: Makefile [hunk 5]
```diff
@@ -124,24 +123,27 @@ cluster-clean: $(KUBECTL)
...
 	fi
 
 cluster-sync-handler: cluster-sync-resources $(local_handler_manifest)
-	IMAGE_REGISTRY=localhost:$(shell $(CLI) ports registry | tr -d '\r') \
-				   make push-handler
-	# Temporary until image is updated with provisioner that sets this field
-	# This field is required by buildah tool
-	$(SSH) node01 'sudo sysctl -w user.max_user_namespaces=1024'
+	if [[ "$$KUBEVIRT_PROVIDER" =~ ^okd-.*$$ ]]; then \
+		IMAGE_REGISTRY=localhost:$(shell $(CLI) ports --prefix=$$KUBEVIRT_PROVIDER --container-name=cluster registry | tr -d '\r') \
```
Comment (phoracek) [inline Makefile]: Is this a workaround of a known issue?

Comment (qinqon) [reply Makefile]: It was failing on okd, and at kubernetes-nmstate we don't use buildah.

Comment (phoracek) [reply Makefile]: How is it related to buildah?

File: hack/install-nm.sh (hunk 3 above)
Comment (phoracek) [reply hack/install-nm.sh]: Still pending

**Comment by qinqon** [comment]
Looks like container name to get the ports is "nmstate_kubernetes-nmstate_standard-check-pr-cluter"

File: Makefile (hunk 5 above)
Comment (qinqon) [reply Makefile]: No clue, it was there at kubevirtci dind't do the research.

**Comment by qinqon** [comment]
ci test please

File: test/e2e/default_bridged_network_test.go [hunk 6]
```diff
@@ -22,7 +22,7 @@ import (
 
 // FIXME: We have to fix this test https://github.com/nmstate/kubernetes-nmstate/issues/192
 var _ = PDescribe("NodeNetworkConfigurationPolicy default bridged network", func() {
-	createBridgeOnTheDefaultInterface := nmstatev1alpha1.State(`interfaces:
+	createBridgeOnTheDefaultInterface := nmstatev1alpha1.State(fmt.Sprintf(`interfaces:
```
Comment (phoracek) [inline test/e2e/default_bridged_network_test.go]: This should be in states.go too (not related to the PR)

**Comment by phoracek** [comment]
ci test please

**Comment by qinqon** [comment]
ci test please

**Comment by qinqon** [comment]
3 failures to go!

**Comment by qinqon** [comment]
Only the rollback test is failing, looks like is not able to retrieve the pod names to move vlan-filtering, locally this is passing.

File: Makefile [hunk 7]
```diff
@@ -124,24 +130,28 @@ cluster-clean: $(KUBECTL)
...
 		$(KUBECTL) apply -f deploy/openshift/; \
 	fi
 
 cluster-sync-handler: cluster-sync-resources $(local_handler_manifest)
-	IMAGE_REGISTRY=localhost:$(shell $(CLI) ports registry | tr -d '\r') \
-				   make push-handler
-	# Temporary until image is updated with provisioner that sets this field
-	# This field is required by buildah tool
-	$(SSH) node01 'sudo sysctl -w user.max_user_namespaces=1024'
+	docker ps
```
Comment (phoracek) [inline Makefile]: please drop this

File: hack/install-nm.sh [hunk 8]
```diff
@@ -1,13 +1,16 @@
...
+script_dir=$(dirname "$(readlink -f "$0")")
+ssh=$script_dir/../kubevirtci/cluster-up/ssh.sh
 
-node01_id=$(docker ps |grep node01 |awk '{print $1}')
+if [[ "$KUBEVIRT_PROVIDER" =~ k8s ]]; then
+    echo 'Install NetworkManager on the node'
+    $ssh node01 -- sudo yum install -y NetworkManager
+    $ssh node01 -- sudo systemctl daemon-reload
+    $ssh node01 -- sudo systemctl restart NetworkManager
+    $ssh node01 -- sudo systemctl restart NetworkManager
```
Comment (phoracek) [inline hack/install-nm.sh]: this line is duplicated

File: test/e2e/default_bridged_network_test.go [hunk 9]
```diff
@@ -43,43 +44,43 @@ var _ = Describe("NodeNetworkConfigurationPolicy default bridged network", func(
   - name: brext
     type: linux-bridge
     state: absent
-`)
+`, *primaryNic))
 	Context("when there is a default interface with dynamic address", func() {
 		addressByNode := map[string]string{}
 		BeforeEach(func() {
-			By("Check eth0 is the default route interface and has dynamic address")
+			By(fmt.Sprintf("Check %s is the default route interface and has dynamic address", *primaryNic))
```
Comment (phoracek) [inline test/e2e/default_bridged_network_test.go]: Man, someone should create `Byf(msg, args...)` ...

Comment (phoracek) [reply test/e2e/default_bridged_network_test.go]: https://github.com/nmstate/kubernetes-nmstate/issues/212

File: test/e2e/find-bridge-vlan-centos-8.sh [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash -e
```
Comment (phoracek) [inline test/e2e/find-bridge-vlan-centos-8.sh]: The script is named "find-bridge-vlan-centos-8". Not sure I understand the "find" part there. What about `check-has-vlans-el8.sh`? It look very much alike `func hasVlans(node string, connection string, minVlan int, maxVlan int) AsyncAssertion`.

File: test/e2e/find-bridge-vlan-centos-8.sh [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+vlan_min=$3
+vlan_max=$4
+vlans_out=/tmp/vlans.out
+
+echo "Dumping bridge vlans to $vlans_out"
+./kubevirtci/cluster-up/ssh.sh $node -- sudo bridge vlan show > $vlans_out
+
+echo "Checking vlan range $vlan_min-$vlan_max at $connection"
+for vlan in $(seq $vlan_min $vlan_max); do
+    if ! cat $vlans_out|grep $connection -A 1 |grep " *$vlan *" > /dev/null; then
```
Comment (phoracek) [inline test/e2e/find-bridge-vlan-centos-8.sh]: `     if ! cat $vlans_out | grep $connection -A 1 | grep " *$vlan *" > /dev/null; then `

File: test/e2e/find-bridge-vlan-flags-centos-8.sh [hunk 12]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -e
```
Comment (phoracek) [inline test/e2e/find-bridge-vlan-flags-centos-8.sh]: `get-bridge-vlan-flags-el8.sh`?

File: test/e2e/default_bridged_network_test.go (hunk 9 above)
Comment (qinqon) [reply test/e2e/default_bridged_network_test.go]: Agree, thanks for the issue.

**Comment by phoracek** [comment]
@qinqon there seems to be a real bug, maybe even in the rollback code:

```
, rollback error: failed to execute nmstatectl rollback: 'exit status 65' 'ERROR rolling back change: No checkpoint specified or found
```

**Comment by phoracek** [comment]
Let's tackle the test later. Basic tests look fine. LGTM

**Pull request merged**